test/unit/quad_stamp_test.rb

```diff
@@ -1,487 +1,489 @@
-                'Quadrant 2' => 'DN123T',
-                'Quadrant 3' => 'Empty',
-                'Quadrant 4' => 'Empty'
+                "Quadrant 1" => "Empty",
+                "Quadrant 2" => "DN123T",
+                "Quadrant 3" => "Empty",
+                "Quadrant 4" => "Empty"
               },
               size: 384
             )
 
           # need a factory that creates a v2 plate with parents & custom metadata referencing those parents
           # 2 robots
 
-          Sequencescape::Api::V2::Plate.stubs(:where).with(barcode: 'DN456S').returns([child_plate])
+          Sequencescape::Api::V2::Plate.stubs(:where).with(barcode: "DN456S").returns([child_plate])
 
           @input_params = {
-            user_barcode: '123',
+            user_barcode: "123",
             instrument_barcode: @instrument.barcode.to_s,
             instrument_process: @instrument.instrument_processes.first.id.to_s,
             robot: {
               p4: {
-                bed: '4',
-                plate: 'DN123T'
+                bed: "4",
+                plate: "DN123T"
               },
               p5: {
-                bed: '',
-                plate: ''
+                bed: "",
+                plate: ""
               },
               p6: {
-                bed: '',
-                plate: ''
+                bed: "",
+                plate: ""
               },
               p8: {
-                bed: '',
-                plate: ''
+                bed: "",
+                plate: ""
               },
               p9: {
-                bed: '9',
-                plate: 'DN456S'
+                bed: "9",
+                plate: "DN456S"
               }
             }
           }
 
-          api = TestSequencescapeApi.new({ 'DN456S' => [TestSearchResult.new('DN123T')] })
+          api = TestSequencescapeApi.new({ "DN456S" => [TestSearchResult.new("DN123T")] })
 
           @old_delayed_job_count = Delayed::Job.count
           @bed_layout_verification =
             Verification::QuadStampPlate::Bravo.new(
               instrument_barcode: @input_params[:instrument_barcode],
               scanned_values: @input_params[:robot],
-              api: api
+              api:
             )
-          User.expects(:login_from_user_code).at_least(0).with(@input_params[:user_barcode]).returns('abc')
+          User.expects(:login_from_user_code).at_least(0).with(@input_params[:user_barcode]).returns("abc")
 
           @bed_layout_verification.validate_and_create_audits?(@input_params)
           @new_delayed_job_count = Delayed::Job.count
         end
 
-        should 'return an error' do
+        should "return an error" do
           error_message = 'The barcode in bed P4 doesn\'t match the plate in Quadrant 1 on the destination plate.'
-          assert_includes @bed_layout_verification.errors.values.flatten, error_message
+
+          assert_includes @bed_layout_verification.errors[:base], error_message
```

Comment:
This is because of the changes in the errors structure while upgrading, right?

Reply:
Yes, errors.values is no longer supported